pkg/services/provisioning/values/values.go

```diff
@@ -151,14 +154,15 @@ func (val *StringMapValue) Value() map[string]string {
 	return val.value
 }
 
-// tranformInterface tries to transform any interface type into proper value with env expansion. It travers maps and
+// transformInterface tries to transform any interface type into proper value with env expansion. It travers maps and
 // slices and the actual interpolation is done on all simple string values in the structure. It returns a copy of any
-// map or slice value instead of modifying them in place.
-func tranformInterface(i interface{}) interface{} {
+// map or slice value instead of modifying them in place and also return value without interpolation but with converted
+// type as a second value.
+func transformInterface(i interface{}) (interface{}, interface{}) {
```

Comment:
Returning 2 values so that both raw value and value which was interpolated has the same types and there is no discrepancy. Also non string keys will be dropped (should not happen but yaml technically allows non string keys afaik) which would also create a difference in raw vs interpolated value.